sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1198,8 +1209,34 @@ def process(self, element, *schema_side_inputs):
       return self._flush_all_batches()
 
   def finish_bundle(self):
+    current_millis = int(time.time() * 1000)
+    if BigQueryWriteFn.LATENCY_LOGGING_LOCK.acquire(False):
```

Comment:
acquire with the false parameter can skip this section entirely, if another thread has the lock. So if that happens we won't record current_millis into the histogram

Reply:
`current_millis` is only for checking whether the enough amount of time passed after the last log reporting time. `acquire` with the false parameter was intentional to minimize performance impact for log reporting. Please see the other comment above.